junit-jupiter-api/src/main/java/org/junit/jupiter/api/Assertions.java

```diff
@@ -3070,7 +3070,8 @@ public static <T extends Throwable> T assertThrowsExactly(Class<T> expectedType,
 	 * <p>If you do not want to perform additional checks on the exception instance,
 	 * ignore the return value.
 	 *
-	 * <p>Fails with the supplied failure {@code message}.
+	 * <p>Fails with the supplied failure {@code message}. That message is <em>not</em>
+	 * the expected message for the thrown exception.
```

Comment:
Ok, I have adjusted the documentation with this now. For the methods with `Supplier` I slightly modified the first sentence since there is no `message` parameter.